### ISSUE TITLE
Ask build tools and IDEs to upgrade to junit-platform-launcher 1.6.x

### ISSUE THREAD

**Issue by marcphilipp** [junit-team/junit5#2028]
## Overview

In #1909 we've improved error reporting for engine discovery/execution failures. The relevant changes are in `DefaultLauncher` which is contained in the `junit-platform-launcher` artifact. Since most build tools and IDEs bundle that artifact we need to reach out to them and ask them to update to Jupiter/Vintage 5.6 (Platform 1.6).

## Projects

List of projects and contacts to ping.

- [x] Ant, `JUnitLauncher` task doesn't bundle JUnit artifacts
- [x] Eclipse, @noopur2507
- [x] [Gradle](https://github.com/gradle/gradle/pull/12489), @marcphilipp 
- [x] IntelliJ IDEA, @akozlova
- [x] ~Maven Surefire/Failsafe, @sormuras~ works with Surefire 3.x
- [x] Netbeans
- [x] VS Code, _work-in-progress_ ... [vscode-java-test #975](https://github.com/microsoft/vscode-java-test/issues/975)

## Deliverables

- [x] All of the above build tools and IDEs have been upgraded

**Comment by akozlova** [comment]
IntelliJ doesn't bundle junit-platform-launcher anymore (about a year or even more), it downloads the launcher on tests start according to the version of the api used in the user's project.

**Comment by marcphilipp** [comment]
@akozlova Thanks for the info! In that case, we're fine where IDEA is concerned.

**Comment by noopur2507** [comment]
Eclipse bug report to move to JUnit 5.6: [bug 558873](https://bugs.eclipse.org/bugs/show_bug.cgi?id=558873).

**Comment by sormuras** [comment]
Surefire issue: https://issues.apache.org/jira/browse/SUREFIRE-1750

**Comment by bmuskalla** [comment]
Gradle has been upgraded to the 1.6.0 launcher which will be released as part of Gradle 6.4

**Comment by marcphilipp** [comment]
@bmuskalla Thanks for letting us know! 👍

**Comment by sormuras** [comment]
Created https://issues.apache.org/jira/browse/SUREFIRE-1764 to update Surefire's dependency.

**Comment by Tibor17** [comment]
In Surefire we are resolving the version of `junit-platform-launcher` in runtime. So we only need to fix it and not to upgrade it. Upgrading it would mean either breaking the users or forcing them to upgrade their POMs as well. Pls discuss it.

**Comment by sormuras** [comment]
> In Surefire [...] Pls discuss it.

Let's move the Surefire-related discussion to https://issues.apache.org/jira/browse/SUREFIRE-1764 -- this issue collecting the state of "all" build tools and IDEs.

**Comment by marcphilipp** [comment]
@sormuras will create tickets for the remaining projects from the issue description.

**Comment by sormuras** [comment]
Ant is "always up-to-date" as it doesn't bundle any artifact from JUnit 5 -- users are in charge to update their dependencies.

**Comment by sormuras** [comment]
Netbeans includes JUnit 5.6 since Feb/Mar 2020, i.e. Netbeans 12: https://github.com/apache/netbeans/pull/1977

**Comment by sormuras** [comment]
Seems like @jlink is [working on VS Code](https://github.com/microsoft/vscode-java-test/commits/master/java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/searcher/JUnit5TestSearcher.java) lately -- is VS Code already on Platform 1.6 or higher, Johannes?

**Comment by sormuras** [comment]
Moved to 5.7 Backlog as not all IDEs/build tools are on JUnit Platform 1.6, resp. Jupiter 5.6.

**Comment by jlink** [comment]
@sormuras All I did is to enable VS Code to run other engine tests from an editor.

Under the hood VS Code is using some Eclipse-related library to start JUnit 5 tests in a very hacky way (i.e. through a faulty command line). It basically works with no engine correctly, with Jupiter it does the simple things right.

I offered to support them when migrating to use junit-platform-launcher. As of today they have not taken me up on my offer.
